From 6bf8a942bfd0cebbef55e17531382a0e2746eef0 Mon Sep 17 00:00:00 2001
From: Martin Kolman <mkolman@redhat.com>
Date: Mon, 15 Jun 2015 16:48:24 +0200
Subject: [PATCH] Add a README (#1194155)

Add a short readme that contains instructions on how to re-enable
Firstboot and a deprecation notice. The deprecation notice includes
links to Firstboot plugin -> Anaconda/Initial Setup addon porting
resources.

Resolves: rhbz#1194155
---
 MANIFEST.in    |  2 +-
 README.txt     | 20 ++++++++++++++++++++
 2 files changed, 21 insertions(+), 1 deletion(-)
 create mode 100644 README.txt

diff --git a/MANIFEST.in b/MANIFEST.in
index 2723803..26babb5 100644
--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -1,4 +1,4 @@
-include Makefile firstboot.spec
+include Makefile firstboot.spec README.txt
 recursive-include init firstboot
 recursive-include po *.po *.pot Makefile
 recursive-include progs firstboot
diff --git a/README.txt b/README.txt
new file mode 100644
index 0000000..0693910
--- /dev/null
+++ b/README.txt
@@ -0,0 +1,20 @@
+== How to re-enable Firstboot after first boot ==
+To re-enable Firstboot, you need to do the following:
+
+ 1. rm /etc/sysconfig/firstboot
+ 2. systemctl enable firstboot-graphical.service
+
+NOTE: /etc/reconfigSys is now (RHEL7) ignored.
+
+== Advisory to porting Firstboot plugins ==
+
+The legacy Firstboot tool is no longer in development and all current Firstboot users are advised to port their legacy Firstboot plugins to Anaconda addons, which don't have most of the limitations imposed by the Firstboot architecture.  Also unlike Firstboot plugins, Anaconda addons can be run both of during installation (by Anaconda), and after installation (by Initial Setup).
+
+A comprehensive Anaconda Addon Development Guide is available:
+http://rhinstaller.github.io/anaconda-addon-development-guide/
+
+As well as an example "Hello world" addon:
+https://github.com/rhinstaller/hello-world-anaconda-addon
+
+Also the Kdump project recently successfully made the transition from a Firstboot plugin to an Anaconda addon and could be used as good reference example for a more advanced addon project:
+https://github.com/daveyoung/kdump-anaconda-addon
-- 
2.4.2

